src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+            if (!unknownGroups.isEmpty()){
+                throw new BadArgumentValue("group", "Unknown annotation group in " + unknownGroups + ". Known groups are " + Arrays.toString(AnnotationGroup.values()) );
+            }
+
+        }
+
+        static AnnotationManager ofSelected(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            final List<String> groups = new ArrayList<>(annotationGroupsToUse);//make copy
+            final List<String> annots = new ArrayList<>(annotationsToUse);//make copy
+            annots.removeAll(annotationsToExclude);
+            return new AnnotationManager(groups, annots);
+        }
+
+        static AnnotationManager ofAll(final List<String> annotationsToExclude){
+            final List<String> groups = getAllAnnotationGroupNames();
+            final List<String> annots = getAllAnnotationNames();
+            annots.removeAll(annotationsToExclude);
+            return new AnnotationManager(groups, annots);
+        }
+
+        private static List<String> getAllAnnotationNames() {
+            final Set<VariantAnnotation> union = Sets.union(new HashSet<>(makeAllGenotypeAnnotations()), new HashSet<>(AnnotationManager.makeAllInfoFieldAnnotations()));
+            return union.stream().map(a -> a.getClass().getSimpleName()).collect(Collectors.toList());
+        }
+
+        public static List<String> getAllAnnotationGroupNames() {
+            return Stream.of(AnnotationGroup.values()).map(v -> v.name()).collect(Collectors.toList());
+        }
+
+        public List<InfoFieldAnnotation> createInfoFieldAnnotations() {
+            final List<InfoFieldAnnotation> all = makeAllInfoFieldAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<InfoFieldAnnotation> makeAllInfoFieldAnnotations() {
+            return Arrays.asList(
+                    new InbreedingCoeff(Standard, ActiveRegionBased),
+                    new Coverage(Standard, ActiveRegionBased),
+                    new ReadPosRankSumTest(Standard, ActiveRegionBased),
+                    new BaseQualityRankSumTest(Standard, ActiveRegionBased),
+                    new MappingQualityRankSumTest(Standard, ActiveRegionBased),
+                    new ChromosomeCounts(Standard, ActiveRegionBased),
+                    new StrandOddsRatio(Standard, ActiveRegionBased),
+                    new FisherStrand(Standard, ActiveRegionBased),
+                    new RMSMappingQuality(Standard, ActiveRegionBased),
+                    new QualByDepth(Standard, ActiveRegionBased),
+                    new MappingQualityZero(ActiveRegionBased),
+                    new SampleList(),
+                    new ClippingRankSumTest(ActiveRegionBased),
+                    new LikelihoodRankSumTest(ActiveRegionBased),
+                    new TandemRepeat(ActiveRegionBased),
+                    new GenotypeSummaries(ActiveRegionBased)
```

Comment:
Why have hardcoded lists of annotations that need to be maintained when we can just discover all annotations dynamically in ~2 lines of code?

            final ClassFinder finder = new ClassFinder();
            return finder.find("org.broadinstitute.hellbender", InfoFieldAnnotation.class).getConcreteClasses();

Hardcoding annotations in the engine is also very unfriendly to projects that use hellbender as a dependency.